accurest-core/src/main/groovy/io/codearte/accurest/builder/SpockMethodBodyBuilder.groovy

```diff
@@ -20,46 +25,81 @@ class SpockMethodBodyBuilder {
 	}
 
 	void appendTo(BlockBuilder blockBuilder) {
-		blockBuilder.startBlock()
-		blockBuilder.addLine('given:').startBlock()
-		blockBuilder.addLine('def request = given()')
-		blockBuilder.indent()
-		stubDefinition.request.headers?.collect { Header header ->
-			blockBuilder.addLine(".header('${header.name}', '${header.serverValue}')")
-		}
-		if (stubDefinition.request.body) {
-			String matches = new JsonOutput().toJson(stubDefinition.request.body.serverValue)
-			blockBuilder.addLine(".body('$matches')")
-		}
+		Request request = stubDefinition.request
+		Response response = stubDefinition.response
+		blockBuilder.with {
+			startBlock()
+			addLine('given:').startBlock()
+			addLine('def request = given()')
+			indent()
+			request.headers?.collect { Header header ->
+				addLine(".header('${header.name}', '${header.serverValue}')")
+			}
+			if (request.body) {
+				String matches = new JsonOutput().toJson(request.body.serverValue)
+				addLine(".body('$matches')")
+			}
 
-		blockBuilder.unindent().endBlock().addEmptyLine()
+			unindent().endBlock().addEmptyLine()
 
-		blockBuilder.addLine('when:').startBlock()
-		blockBuilder.addLine('def response = given().spec(request)')
-		blockBuilder.indent()
-		blockBuilder.addLine(".${stubDefinition.request.method.serverValue.toLowerCase()}(\"$stubDefinition.request.url.serverValue\")")
-		blockBuilder.unindent().endBlock().addEmptyLine()
+			addLine('when:').startBlock()
+			addLine('def response = given().spec(request)')
+			indent()
 
-		blockBuilder.addLine('then:').startBlock()
-		blockBuilder.addLine("response.statusCode == $stubDefinition.response.status.serverValue")
+			String url = buildUrl(request)
+			String method = request.method.serverValue.toLowerCase()
 
-		stubDefinition.response.headers?.collect { Header header ->
-			blockBuilder.addLine("response.header('$header.name') == '$header.serverValue'")
-		}
-		if (stubDefinition.response.body) {
-			blockBuilder.endBlock()
-			blockBuilder.addLine('and:').startBlock()
-			blockBuilder.addLine('def responseBody = new JsonSlurper().parseText(response.body.asString())')
-			def responseBody = stubDefinition.response.body.serverValue
-			if (responseBody instanceof List) {
-				processArrayElements(responseBody, "", blockBuilder)
-			} else {
-				processMapElement(responseBody, blockBuilder, "")
+			blockBuilder.addLine(/.${method}("$url")/)
+			unindent().endBlock().addEmptyLine()
+
+			addLine('then:').startBlock()
+			addLine("response.statusCode == $response.status.serverValue")
+
+			response.headers?.collect { Header header ->
+				addLine("response.header('$header.name') == '$header.serverValue'")
+			}
+			if (response.body) {
+				endBlock()
+				addLine('and:').startBlock()
+				addLine('def responseBody = new JsonSlurper().parseText(response.body.asString())')
```

Comment:
Since you're already changing this section. Can you do a check of `contentType` in a request and if it is ending with `json` then you can use this `def responseBody = new JsonSlurper().parseText(response.body.asString())'`. WDYT?